public/app/dsp-processor.js

```
@@ -322,66 +353,171 @@ class DspProcessor extends AudioWorkletProcessor {
+  //    γ(k)  = a posteriori SNR = |Y(k)|² / σ²_n(k)
+  //    v(k)  = ξ(k) * γ(k) / (1 + ξ(k))
+  //    E1(v) = exponential integral ≈ approximated via rational function
+  _logMMSE(mag) {
     const nBins    = this.numBins;
-    const amount   = this.params.nrSpectralSub / 100;
-    const floor    = Math.pow(10, this.params.nrFloor / 20);
-    const alphaDD  = 0.92 + (this.params.nrSmoothing / 100) * 0.07; // 0.92–0.99
-    const beta     = 1.5 + amount; // over-subtraction factor 1.5–2.5
-    const prevMag  = this.prevMagEstimate;
+    const amount   = this.params.nrAmount / 100;           // 0–1 user control
+    const subAlpha = this.params.nrSpectralSub / 100;      // spectral sub blend
```

ESLint

[error] 'subAlpha' is assigned a value but never used.

(no-unused-vars)